src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -1276,17 +1276,17 @@ internal static object[] GetCustomAttributes(RuntimeType type, RuntimeType caTyp
                 return attributes;
             }
 
-            List<object> result = new List<object>();
+            RuntimeType.ListBuilder<object> result = new RuntimeType.ListBuilder<object>();
             bool mustBeInheritable = false;
             bool useObjectArray = (caType == null || caType.IsValueType || caType.ContainsGenericParameters);
-            Type arrayType = useObjectArray ? typeof(object) : caType;
+            RuntimeType arrayType = useObjectArray ? typeof(object) as RuntimeType : caType;
```

Comment:
I generally think of `as` being used when we think it's possible the cast may fail, and want a null instead of an exception.  But this should never fail, so can we use `(RuntimeType)typeof(object)` instead?  It's also more consistent with what's used elsewhere in this method already.

Reply:
Absolutely - done!